### PR TITLE
Add OCSP support

### PR DESCRIPTION
**Pull request by carl-wallace**
Same as earlier PR with mods to align with recent upstream changes.

### CODE REVIEW

File: ocsp/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[package]
+name = "x509-ocsp"
```
Comment (tarcieri) [inline ocsp/Cargo.toml:2]: It'd probably be good if the directory name matched the crate name (i.e. `x509-ocsp`)

I should probably rename the `x509` directory to `x509-cert` for consistency as well.

File: ocsp/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# [RustCrypto]: X.509
```
Comment (tarcieri) [inline ocsp/README.md:1]:
```suggestion
# [RustCrypto]: X.509 Online Certificate Status Protocol (OCSP)
```

File: ocsp/Cargo.toml (hunk 1 above)
Comment (carl-wallace) [reply ocsp/Cargo.toml]: I changed the ocsp folder.

File: ocsp/README.md (hunk 2 above)
Comment (carl-wallace) [reply ocsp/README.md]: Fixed

File: x509-ocsp/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+[package]
+name = "x509-ocsp"
+version = "0.0.1"
+description = """
+Pure Rust implementation of the X.509 Internet Public Key Infrastructure
+Online Certificate Status Protocol - OCSP formats as described in RFC 6960
+"""
+authors = ["RustCrypto Developers"]
+license = "Apache-2.0 OR MIT"
+repository = "https://github.com/RustCrypto/formats/tree/master/x509"
```
Comment (tarcieri) [inline x509-ocsp/Cargo.toml:10]:
```suggestion
repository = "https://github.com/RustCrypto/formats/tree/master/x509-ocsp"
```

File: x509-ocsp/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+//! The ocsp module features encoders and decoders for the structures defined in [RFC 6960](https://datatracker.ietf.org/doc/html/rfc6960).
+#![no_std]
+
+pub mod ocsp;
```
Comment (tarcieri) [inline x509-ocsp/src/lib.rs:4]: This module seems somewhat superfluous. I'd suggest putting the contents of `ocsp.rs` in `lib.rs` until such time as there's actually enough code to have more than one module.

**Comment by tarcieri** [comment]
Nits aside, this looks mostly ready to go

File: x509-ocsp/Cargo.toml (hunk 3 above)
Comment (carl-wallace) [reply x509-ocsp/Cargo.toml]: fixed

File: x509-ocsp/src/lib.rs (hunk 4 above)
Comment (carl-wallace) [reply x509-ocsp/src/lib.rs]: moved code to lib.rs. I tend to doubt more code lands here, client code ought land elsewhere I think. this is just encoders/decoders.

**Review by tarcieri — APPROVED** [review]

**Pull request merged**